apps/opengl_demo/main.cpp

```diff
@@ -97,21 +83,21 @@ std::string run_opengl_filter_from_texture_to_texture(GLuint input_texture_id, G
     // Create halide input buffer and tell it to use the existing GPU
     // texture.  No need to allocate memory on the host since this simple
     // pipeline will run entirely on the GPU.
-    auto input_buf = create_buffer(width, height);
-    halide_opengl_wrap_texture(nullptr, &input_buf, input_texture_id);
+    auto input_buf = create_buffer(nullptr, width, height);
+    halide_opengl_wrap_texture(nullptr, input_buf.raw_buffer(), input_texture_id);
```

Comment:
Because it's messing with the raw buffer fields in a way not managed by the Buffer class. I want to fix that by adding this wrapping functionality to the buffer class, but for now I want to highlight that this is a bit dodgy.